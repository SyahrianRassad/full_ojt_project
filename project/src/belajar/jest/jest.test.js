import { render, screen, cleanup } from "@testing-library/react"
import renderer from 'react-test-renderer'
import Jest from './jest'

afterEach(()=>{
    cleanup()
})

test('should render jest component', ()=>{
    const data = {id:1, title:"wash dishes", completed: true}
    render(<Jest data={data}/>);
    //test 1
    const jestElement = screen.getByRole('role-1'); //mengambil role
    expect(jestElement).toBeInTheDocument();
    expect(jestElement).toHaveTextContent('Hello World!'); //cek isi content
})

test('should render jest component', ()=>{
    const data = {id:1, title:"wash dishes", completed: false}
    render(<Jest data={data}/>);

    const jestElement2 = screen.getByRole('todo-1');
    expect(jestElement2).toHaveTextContent('wash dishes')
    expect(jestElement2).not.toContainHTML('<strike>')
})

test('should render jest component', ()=>{
    const data = {id:2, title:"wash car", completed: true}
    render(<Jest data={data}/>);

    const jestElement2 = screen.getByRole('todo-2');
    expect(jestElement2).toHaveTextContent('wash car');
    expect(jestElement2).not.toContainHTML('<strike>')
})

test('maches snapshoot', ()=>{
    const data = {id:2, title:"wash car", completed: true}
    const tree = renderer.create(<Jest data={data}/>).toJSON
    expect(tree).toMatchSnapshot();
})