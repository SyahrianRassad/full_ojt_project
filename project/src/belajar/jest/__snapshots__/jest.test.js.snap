// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maches snapshoot 1`] = `[Function]`;
